tmc-langs-r/src/test/java/fi/helsinki/cs/tmc/langs/r/RPluginTest.java

```diff
@@ -0,0 +1,203 @@
+    private Path simpleSomeTestsFailProject;
+    private Path simpleSourceCodeErrorProject;
+
+    @Before
+    public void setUp() {
+        plugin = new RPlugin();
+
+        simpleAllTestsPassProject = TestUtils.getPath(getClass(),
+                "simple_all_tests_pass");
+        simpleSomeTestsFailProject = TestUtils.getPath(getClass(),
+                "simple_some_tests_fail");
+        simpleSourceCodeErrorProject = TestUtils.getPath(getClass(),
+                "simple_source_code_error");
+    }
+
+    @After
+    public void tearDown() {
+        removeAvailablePointsJson(simpleAllTestsPassProject);
+        removeResultsJson(simpleAllTestsPassProject);
+        removeResultsJson(simpleSomeTestsFailProject);
+        removeResultsJson(simpleSourceCodeErrorProject);
+    }
+
+    private void removeResultsJson(Path projectPath) {
+        File resultsJson = new File(projectPath.toAbsolutePath().toString()
+                + "/.results.json");
+        resultsJson.delete();
+    }
+
+    private void removeAvailablePointsJson(Path projectPath) {
+        File availablePointsJson = new File(projectPath.toAbsolutePath().toString()
+                + "/.available_points.json");
+        availablePointsJson.delete();
+    }
+
+    private void testResultAsExpected(TestResult result, boolean successful, String name,
+                                      String[] points) {
+        assertEquals(successful, result.isSuccessful());
+        assertEquals(name, result.getName());
+        assertArrayEquals(points, result.points.toArray());
+    }
+
+    @Test
+    public void testGetTestCommand() {
+        String[] command = new String[]{"Rscript"};
+        String[] args;
+        if (SystemUtils.IS_OS_WINDOWS) {
+            args = new String[]{"-e", "\"library('tmcRtestrunner');run_tests()\""};
+        } else {
+            args = new String[]{"-e", "library(tmcRtestrunner);run_tests()"};
+        }
+        String[] expectedCommand = ArrayUtils.addAll(command, args);
+        Assert.assertArrayEquals(expectedCommand, plugin.getTestCommand());
+    }
+
+    @Test
+    public void testGetAvailablePointsCommand() {
+        String[] command = new String[]{"Rscript"};
+        String[] args;
+        if (SystemUtils.IS_OS_WINDOWS) {
+            args = new String[]{"-e", "\"library('tmcRtestrunner');run_available_points()\""};
+        } else {
+            args = new String[]{"-e", "library(tmcRtestrunner);run_available_points()"};
+        }
+        String[] expectedCommand = ArrayUtils.addAll(command, args);
+        Assert.assertArrayEquals(expectedCommand, plugin.getAvailablePointsCommand());
+    }
+
+    @Test
+    public void testGetPluginName() {
+        assertEquals("r", plugin.getLanguageName());
+    }
+
+    @Test
+    public void testScanExercise() {
+        plugin.scanExercise(simpleAllTestsPassProject, "main.R");
+        File availablePointsJson = new File(simpleAllTestsPassProject.toAbsolutePath().toString()
```

Comment:
This might be problematic in Windows.

How about `Files.exists(simpleAllTestsPassProject.resolve(".available_points.json"))`?

Also, it is not a good idea to use Files in new code. I'd always prefer to use the newer alternative `java.nio.Path` instead.

Reply:
Just to clarify: `java.nio.Files` is fine (it's part of the new IO library), it's `java.io.File` that should be avoided and replaced with `java.nio.Path`.